common/lib/dependabot/git_metadata_fetcher.rb

```diff
@@ -72,8 +72,14 @@ def fetch_upload_pack_for(uri)
       response = fetch_raw_upload_pack_for(uri)
       return response.body if response.status == 200
 
-      response_with_git = fetch_raw_upload_pack_with_git_for(uri)
-      return response_with_git.body if response_with_git.status == 200
+      # if git is installed, try fetching with it in case something we need is configured in .gitconfig (like creds)
```

Comment:
Approach 1: gate the fallback at the callsite... it's more clear exactly what/why we're doing this